.github/workflows/valgrind.yml

```diff
@@ -0,0 +1,41 @@
+# Syntax reference https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions
+# Environment reference https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners
+name: valgrind
+
+on: [push, pull_request]
+
+jobs:
+  build:
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+      fail-fast: false # not worthwhile...
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install missing software on ubuntu
+        if: matrix.os == 'ubuntu-latest'
+        run: |
+          sudo apt-get update
+          sudo apt-get install libxml2-utils
+          sudo apt-get install z3 libz3-dev
+          sudo apt-get install valgrind
+          cp externals/z3_version_old.h externals/z3_version.h
+
+      - name: Build cppcheck
+        run: |
+          make clean
+          cp externals/z3_version_old.h externals/z3_version.h
+          make -j$(nproc) USE_Z3=yes HAVE_RULES=yes
+
+      - name: Build test
+        run: |
+          make -j$(nproc) testrunner USE_Z3=yes HAVE_RULES=yes
+
+      - name: Run valgrind
+        run: |
+          valgrind ./testrunner
```

Comment:
You need to pass `--error-exitcode=1` so it will actually fail on any finding.

I also suggest adding the following options for better and more complete reporting `--error-limit=no --leak-check=full --num-callers=50 --show-reachable=yes --track-fds=yes --track-origins=yes`.

Reply:
@firewave  Yes, but then all CI build would suddenly break which is not desired IMHO.
We should those issues fixed and afterwards enable the check.

Thanks for the suggested command flags!

Reply:
Fine with me. We should hurry though so we don't introduce even more while we are waiting :D

Reply:
We could also reduce the runtime by using `-O1`. That's what usually is suggested and won't mess too much with the stack traces in case of an error.

Reply:
Unfortunately that produces code which valgrind cannot handle

```
vex amd64->IR: unhandled instruction bytes: 0x62 0xF1 0x5D 0x48 0xEF 0xE4 0xC5 0xFB 0x11 0xA5
vex amd64->IR:   REX=0 REX.W=0 REX.R=0 REX.X=0 REX.B=0
vex amd64->IR:   VEX=0 VEX.L=0 VEX.nVVVV=0x0 ESC=NONE
vex amd64->IR:   PFX.66=0 PFX.F2=0 PFX.F3=0
==7492== valgrind: Unrecognised instruction at address 0x7995ee.
```

The version is pretty old though so this might be expected. Will check with the latest version and open a ticket upstream if this persists. 

So back to `-Og`...